notes/design_principles.md

```diff
@@ -0,0 +1,101 @@
+
+On the other hand, repository level events are seen as being in scope for the work to track the versioning of a resource, its audit trail and so forth.  Development work in this area should keep the above in mind and consider whether it is possible to handle both types of event with one set of code.
+
+
+### Identifiers are Resources
+
+Unlike the vanilla `dc:identifier` string, we consider that identifiers are full resources that can have relationships and properties.  In particular, identifiers have a type (e.g. CODEN, ISSN, DOI, Handle, LCCN, Dewey, ...)  which may be only of local significance, and a value.  They can also be deprecated or replaced.  They can be associated with both real world objects, or with digital objects.
+
+```json
+{
+	"id": "http://example.repo/id/1",
+	"type": "bf:Item",
+    "bf:identifiedBy": {
+      "type": "ISBN",
+      "value": ""
+    }
+}
+```
+
+### The Repository cares about derivatives
+
+While some file formats can be easily managed as a single resource with derivatives automatically generated and cached for performance, there are also many other formats where this is not possible.  Audio and Video transcoding is not universally understood, and there may be custom settings applied to individual objects to obtain the best results.
+
+Secondly, derivatives may include representations that involve human interaction, creativity and decision making.  These outputs of activity need to be preserved as they cannot be automatically recreated.  Examples include the cropping of page images to remove the scanning bed, the generation of filmstrip thumbnail images to represent a video, or simply the selection of the most interesting part of the most appropriate representative image from a manuscript.  This implies that the depositor or curator can select this representative part, rather than it being automatically determined by the system.
+
+The relationship between master and derivative should be maintained in the repository, and these files and relationships should be able to be bulk imported.
+
+### Collection information is important
+
+Different features for collections include dates and places of exhibition, provenance information, and so forth.  Many of the features of objects will be useful, and the exact nature of collection description needs to be discussed and decided upon.
+
+### Access patterns are outside of the scope of curators and the model
```

Comment:
It's from the scenarios where the depositor "indicates" that the TIFF will be turned into a JP2, and then that there will be pan and zoom based UI.  That has some implications, none of which I like:

* The depositor has sufficient knowledge of the workings of the system to select JP2 explicitly. That seems unreasonable to expect of random faculty and students.
* The depositor has sufficient knowledge of UI/UX to be able to select that the images should be pan/zoom-able. Also that there is some option for big images to _not_ be pan/zoom-able. Hannah and I could not come up with any even hypothetical scenario where that was desirable.
* Any decisions like these would need to be captured in the model for persistence, and are likely to go out of date very quickly as the technology, UI/UX patterns and best practices, and so on are updated.  Constant remediation of technology specific information across the entire repository seems like something to avoid.